### PR TITLE
chore: bump multiple otel dependencies and fix API changes

### PR DESCRIPTION
**Pull request by GeorgeMac**
I've hijacked the dependency updates for the following dependabot PRs:
- https://github.com/flipt-io/flipt/pull/1322
- https://github.com/flipt-io/flipt/pull/1321
- https://github.com/flipt-io/flipt/pull/1318

This is because the bump in otel dependency was a breaking change.
The metrics portion is considered Beta which lets them break stuff into minor patches.

This updates all the call sites to use the new definitions introduced around version 1.13.
We can close all of those and replace them with this.

### CODE REVIEW

File: internal/metrics/metrics.go [hunk 1]
```diff
@@ -28,29 +26,35 @@ func init() {
...
-	// Counter creates an instrument for recording increasing values.
-	Counter(name string, opts ...instrument.Option) syncint64.Counter
-	// UpDownCounter creates an instrument for recording changes of a value.
-	UpDownCounter(name string, opts ...instrument.Option) syncint64.UpDownCounter
-	// Histogram creates an instrument for recording a distribution of values.
-	Histogram(name string, opts ...instrument.Option) syncint64.Histogram
+// MustInt64Meter is a meter/Meter which panics if it cannot successfully build the
+// requestd counter, upDownCounter or histogram.
+type MustInt64Meter interface {
+	// ounter returns a new instrument identified by name and configured
```
Comment (markphelps) [inline internal/metrics/metrics.go:39]:
```suggestion
	// Counter returns a new instrument identified by name and configured
```

File: internal/metrics/metrics.go [hunk 2]
```diff
@@ -69,38 +73,44 @@ func (m mustSyncInt64InstrumentProvider) UpDownCounter(name string, opts ...inst
...
-	// Counter creates an instrument for recording increasing values.
-	Counter(name string, opts ...instrument.Option) asyncint64.Counter
-	// UpDownCounter creates an instrument for recording changes of a value.
-	UpDownCounter(name string, opts ...instrument.Option) asyncint64.UpDownCounter
-	// Histogram creates an instrument for recording a distribution of values.
-	Gauge(name string, opts ...instrument.Option) asyncint64.Gauge
+// MustFloat64Meter is a meter/Meter which panics if it cannot successfully build the
+// requestd counter, upDownCounter or histogram.
+type MustFloat64Meter interface {
+	// ounter returns a new instrument identified by name and configured
```
Comment (markphelps) [inline internal/metrics/metrics.go:95]:
```suggestion
	// Counter returns a new instrument identified by name and configured
```

**Review by markphelps — APPROVED** [review]
thank you! just 2 minor doc nits

File: internal/metrics/metrics.go (hunk 1 above)
Comment (GeorgeMac) [reply internal/metrics/metrics.go:39]: Thank you. I did a bit of over zealous comment stripping 😂

**Pull request merged**